libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/RouteStateController.kt

```diff
@@ -2,14 +2,22 @@ package com.mapbox.navigation.ui.app.internal.controller
 
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
 import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.RouterOrigin
 import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.extensions.flowRouteAlternativeObserver
 import com.mapbox.navigation.ui.app.internal.Action
 import com.mapbox.navigation.ui.app.internal.State
 import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
 import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.utils.internal.ifNonNull
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
-@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
-class RouteStateController(store: Store) : StateController() {
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class, InternalCoroutinesApi::class)
```

Comment:
The annotation is not necessary, because no experimental APIs are used in this class. 
```suggestion
```

Reply:
`ExperimentalPreviewMapboxNavigationAPI::class` is required because the super class uses it.